### PR TITLE
Reduce & simplify API

### PR DESCRIPTION
**Pull request by radeksimko**
Updated documentation available at https://pkg.go.dev/github.com/hashicorp/terraform-registry-address@v0.0.0-20220610213838-4b195be62a2c

## Follow-up PRs

 - [x] `terraform`: https://github.com/hashicorp/terraform/pull/28338
 - [x] `terraform-schema`: https://github.com/hashicorp/terraform-schema/pull/112
 - [x] `terraform-ls`: https://github.com/hashicorp/terraform-ls/pull/949
 - [x] `terraform-plugin-go`: https://github.com/hashicorp/terraform-plugin-go/pull/200

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,147 @@
...
-`registry.terraform.io/grafana/grafana` or `hashicorp/aws`.
+This module enables parsing, comparison and canonical representation of
+[Terraform Registry](https://registry.terraform.io/) **provider** addresses
+(such as `registry.terraform.io/grafana/grafana` or `hashicorp/aws`)
+and **module** addresses (such as `hashicorp/subnets/cidr`).
 
-The most common source of these addresses outside of Terraform Core
-is JSON representation of state, plan, or schemas as obtained
-via [`hashicorp/terraform-exec`](https://github.com/hashicorp/terraform-exec).
+**Provider** addresses can be found in
```
Comment (bflad) [inline README.md:8]: Do we want to mention the [CLI configuration file](https://www.terraform.io/cli/config/config-file#provider-installation) and [reattach configurations](https://www.terraform.io/plugin/debugging#running-terraform-with-a-provider-in-debug-mode) as well?

File: README.md [hunk 2]
```diff
@@ -1,48 +1,147 @@
...
+
+## Related Libraries
+
+Other libraries which may help with consuming most of the above Terraform
+outputs in automation:
+
+ - [`hashicorp/terraform-exec`](https://github.com/hashicorp/terraform-exec)
+ - [`hashicorp/terraform-json`](https://github.com/hashicorp/terraform-json)
+
+## Usage
```
Comment (bflad) [inline README.md:35]: Would it be worth adding [testable examples](https://go.dev/blog/examples) that would render in Go documentation as well?

File: README.md [hunk 3]
```diff
@@ -1,48 +1,147 @@
...
-the address was produced by an affected version.
+However `NewProvider()` will panic if you pass an empty namespace
+or any placeholder indicating unknown namespace.
 
-If you do not have that context you should parse the string via
-`ParseRawProviderSourceString` and then check `addr.IsLegacy()`.
+```go
+NewProvider(DefaultProviderRegistryHost, "aws", "")  // panic
+NewProvider(DefaultProviderRegistryHost, "aws", "-") // panic
+NewProvider(DefaultProviderRegistryHost, "aws", "?") // panic
```
Comment (bflad) [inline README.md:136]: Are the namespace and provider name parameters flipped here?

**Review by bflad — COMMENTED** [review]
Overall looking good from the plugin SDK/framework perspective 👍

File: README.md (hunk 3 above)
Comment (radeksimko) [reply README.md]: Yes, good catch - thanks!

File: README.md (hunk 2 above)
Comment (radeksimko) [reply README.md:37]: Added.

File: README.md (hunk 1 above)
Comment (radeksimko) [reply README.md:8]: Added.

**Review by jbardin — APPROVED** [review]

**Pull request merged**